test/gomod.collector.test.ts

```diff
@@ -0,0 +1,131 @@
+import { expect } from 'chai';
+import { GomodDependencyCollector } from '../src/collector';
+
+describe('Golang go.mod parser test', () => {
+    const collector:GomodDependencyCollector = new GomodDependencyCollector();
+
+    it('tests valid go.mod', async () => {
+        const deps = await collector.collect(`
+          module github.com/alecthomas/kingpin
+          require (
+            github.com/alecthomas/units v0.0.0-20151022065526-2efee857e7cf
```

Comment:
Will you new implementation work with entry `github.com/davecgh/go-spew v2.1.1+incompatible+version`?

Reply:
Yes, now that we are passing all version strings to API. This should also work. I have added test case covering this.